File: tests/0476_number_complement_test.cpp
#include "0476_number_complement.cpp"
#include <doctest.h>

TEST_CASE("numberComplement")
{
    // setup
    Solution solution = Solution();

    REQUIRE(solution.findComplement(5) == 2);
    REQUIRE(solution.findComplement(2) == 1);
    REQUIRE(solution.findComplement(1) == 0);
    REQUIRE(solution.findComplement(0) == 0);
}

File: tests/0476_number_complement_test.py
import importlib
import pytest

module = importlib.import_module("algorithms.0476_number_complement")


def test_find_complement():
    s = module.Solution()
    assert s.findComplement(5) == 2
    assert s.findComplement(2) == 1
    assert s.findComplement(1) == 0
    assert s.findComplement(0) == 0
